libs/ui/select/brain/src/lib/brn-select.service.ts

```diff
@@ -137,17 +137,20 @@ export class BrnSelectService {
 	}
 
 	private selectOptionByValue(value: unknown) {
+		const options = this._possibleOptions();
+
 		if (value === null || value === undefined) {
-			this.state.update((state) => ({
-				...state,
-				selectedOptions: [],
-				value: this.multiple() ? [] : '',
-			}));
-			return;
+			const nullOrUndefinedOption = options.find((o) => o?.value === value);
```

Comment:
I see what you're saying. Maybe we should add a check for o to be defined just in case?

Reply:
I think it should still be fine right? Because null is only equal to undefined if you do == comparison, if you do === strict comparison it should be false I believe

Reply:
Actually the optional chaining ?. is undefined on a null value. 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

Reply:
Nice catch then!

Reply:
```suggestion
			const nullOrUndefinedOption = options.find((o) => o && o.value === value);
```
What do you guys think of adding this check? @thatsamsonkid @elite-benni